File: 2540-minimum-common-value/2540-minimum-common-value.py
class Solution(object):
    def getCommon(self, nums1, nums2):
        n, m, i, j = len(nums1), len(nums2), 0, 0

        while i < n and j < m:
            if nums1[i] > nums2[j]:
                j += 1
            elif nums1[i] < nums2[j]:
                i += 1
            else:
                return nums2[j]
        
        return -1
        

File: 2540-minimum-common-value/2540-minimum-common-value.cpp
class Solution {
public:
    int getCommon(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size(), m = nums2.size();
        int i = 0, j = 0;

        while(i < n && j < m) {
            if(nums1[i] > nums2[j]) j++;
            else if(nums1[i] < nums2[j]) i++;
            else return nums1[i];
        }

        return -1;
    }
};